packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+  Component,
+  Event,
+  EventEmitter,
+  Host,
+  Prop,
+  h,
+  State,
+} from '@stencil/core';
+
+import {
+  months,
+  days,
+} from '../../../../react-components/src/helpers/options-for-select.js';
+
+import { isFullDate } from '../../../../react-components/src/helpers/validations.js';
+
+@Component({
+  tag: 'va-date',
+  styleUrl: 'va-date.css',
+  shadow: true,
+})
+export class VaDate {
+  /**
+   * Render marker indicating field is required.
+   */
+  @Prop() required: boolean;
+
+  /**
+   * Label for the field.
+   */
+  @Prop() label!: string;
+
+  /**
+   * Used to create unique name attributes for each input.
+   */
+  @Prop() name!: string;
+
+  /**
+   * The error message to render (if any)
+   * This prop should be leveraged to display any custom validations needed for this component
+   */
+  @Prop() error: string;
+
+  /**
+   * Set the `min` value on the year input.
+   */
+  @Prop() minYear: number;
+
+  /**
+   * Set the `max` value on the year input.
+   */
+  @Prop() maxYear: number;
+
+  /**
+   * Set the default date value must be in YYYY-MM-DD format.
+   */
+  @Prop() value: string;
+
+  /**
+   * Fires when the date input loses focus after its value was changed
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  dateChangeEvent: EventEmitter;
+
+  /**
+   * Fires when the date input loses focus
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  dateBlurEvent: EventEmitter;
+
+  @State() month: string;
+  @State() day: string;
+  @State() year: string;
+
+  private minTwoDigits = n => {
+    return (n < 10 ? '0' : '') + n;
+  };
+
+  private handleDateBlurEvent = (event: FocusEvent) => {
+    this.dateBlurEvent.emit(event);
+  };
+
+  private handleDateChangeEvent = (event: Event) => {
+    const target = event.target as HTMLSelectElement | HTMLInputElement;
+    if (target.classList.contains('select-month')) {
+      this.month = target.value;
+    }
+    if (target.classList.contains('select-day')) {
+      this.day = target.value;
+    }
+    if (target.classList.contains('input-year')) {
+      this.year = target.value;
+    }
+    this.dateChangeEvent.emit(event);
```

Comment:
I have left a comment in the code and left the event firing as it currently does. Mainly it starts us on the path of controlling what is a valid date which might differ from team to team and I would like the app teams to control the validation aspect instead of our team.